enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SelectEmbedOptionsStep.tsx

```diff
@@ -0,0 +1,132 @@
+import { useCallback } from "react";
+import { t } from "ttag";
+
+import type { MetabaseColors } from "metabase/embedding-sdk/theme";
+import { colors as defaultMetabaseColors } from "metabase/lib/colors";
+import { Card, Checkbox, Divider, Group, Stack, Text } from "metabase/ui";
+
+import { useSdkIframeEmbedSetupContext } from "../context";
+
+import { DebouncedColorPillPicker } from "./DebouncedColorPillPicker";
+
+export const SelectEmbedOptionsStep = () => {
+  const { experience, settings, updateSettings } =
+    useSdkIframeEmbedSetupContext();
+
+  const { theme } = settings;
+
+  const isQuestionOrDashboardEmbed =
+    settings.dashboardId || settings.questionId;
+
+  const isExplorationEmbed = settings.template === "exploration";
+
+  const updateColor = useCallback(
+    (nextColors: Partial<MetabaseColors>) => {
+      updateSettings({
+        theme: { ...theme, colors: { ...theme?.colors, ...nextColors } },
+      });
+    },
+    [theme, updateSettings],
+  );
+
+  // Titles are only applicable to dashboards and interactive questions
+  const showTitleOption =
+    settings.dashboardId ||
+    (settings.questionId && settings.isDrillThroughEnabled);
```

Comment:
I used this instead of `experience` because of TypeScript: if `settings.questionId` is defined, it implicitly narrows down the type to `QuestionEmbedOptions` which contains `settings.isDrillThroughEnabled`.

Right now we do not have question titles for static questions, but Oisin's PR hopefully changes that soon.

Reply:
> I think we shouldn't couple the download option with drills for charts, they're completely different things. I think we should fix the underlying problem on the SDK then always make these options available at all time regardless of the option value.

@WiNloSt Absolutely agree. This is more of a stop-gap solution to prevent people from being confused why certain options do nothing. Once we add these options to the SDK, we can drop these conditions.